### ISSUE TITLE
[REVIEW]: A short course about fitting models with the `scipy.optimize` module

### ISSUE THREAD

**Issue by whedon** [openjournals/jose-reviews#16]
**Submitting author:** @arokem (<a href="http://orcid.org/0000-0003-0679-1985">Ariel Rokem</a>)
**Repository:** <a href="https://github.com/arokem/scipy-optimize" target ="_blank">https://github.com/arokem/scipy-optimize</a>
**Version:** 0.1
**Editor:** @labarba
**Reviewer:** @oliviaguest, @ThomasA
**Archive:** <a href="http://dx.doi.org/10.5281/zenodo.1304473" target="_blank">10.5281/zenodo.1304473</a>

## Status

[![status](http://jose.theoj.org/papers/25305d3b2e88f1b1d0f98b762ffdca03/status.svg)](http://jose.theoj.org/papers/25305d3b2e88f1b1d0f98b762ffdca03)

Status badge code:

```
HTML: <a href="http://jose.theoj.org/papers/25305d3b2e88f1b1d0f98b762ffdca03"><img src="http://jose.theoj.org/papers/25305d3b2e88f1b1d0f98b762ffdca03/status.svg"></a>
Markdown: [![status](http://jose.theoj.org/papers/25305d3b2e88f1b1d0f98b762ffdca03/status.svg)](http://jose.theoj.org/papers/25305d3b2e88f1b1d0f98b762ffdca03)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/arokem/scipy-optimize" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@oliviaguest & @ThomasA, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/jose-reviews/invitations

The reviewer guidelines are available here: https://jose.theoj.org/about#reviewer_guidelines. Any questions/concerns please let @labarba know.

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSE conflict of interest policy](https://github.com/openjournals/jose/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSE code of conduct](https://jose.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source for this learning module available at the <a target="_blank" href="https://github.com/arokem/scipy-optimize">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of a standard license? (OSI-approved for code, Creative Commons for content)
- [x] **Version:** Does the release version given match the repository release (0.1)?
- [x] **Authorship:** Has the submitting author (@arokem) made substantial contributions to the module? Does the full list of authors seem appropriate and complete?

### Documentation

- [x] **A statement of need:** Do the authors clearly state the need for this module and who the target audience is?
- [x] **Installation instructions:** Is there a clearly stated list of dependencies?
- [x] **Example usage:** Do the authors include examples of how to use the module?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Pedagogy / Instructional design

- [x] **Learning objectives:** Does the module make the learning objectives plainly clear? (We don't require explicitly written learning objectives; only that they be evident from content and design.)
- [x] **Content scope and length:** Is the content substantial for learning a given topic? Is the length of the module appropriate?
- [x] **Pedagogy:** Does the module seem easy to follow? Does it observe guidance on cognitive load? (working memory limits of 7 +/- 2 chunks of information)
- [x] **Content quality:** Is the writing of good quality, concise, engaging? Are the code components well crafted? Does the module seem complete?
- [x] **Instructional design:** Is the instructional design deliberate and apparent? For example, exploit worked-example effects; effective multi-media use; low extraneous cognitive load.

### JOSE paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state the need for this module and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSE conflict of interest policy](https://github.com/openjournals/jose/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSE code of conduct](https://jose.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source for this learning module available at the <a target="_blank" href="https://github.com/arokem/scipy-optimize">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of a standard license? (OSI-approved for code, Creative Commons for content)
- [x] **Version:** Does the release version given match the repository release (0.1)?
- [x] **Authorship:** Has the submitting author (@arokem) made substantial contributions to the module? Does the full list of authors seem appropriate and complete?

### Documentation

- [x] **A statement of need:** Do the authors clearly state the need for this module and who the target audience is?
- [x] **Installation instructions:** Is there a clearly stated list of dependencies?
- [x] **Example usage:** Do the authors include examples of how to use the module?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Pedagogy / Instructional design

- [x] **Learning objectives:** Does the module make the learning objectives plainly clear? (We don't require explicitly written learning objectives; only that they be evident from content and design.)
- [x] **Content scope and length:** Is the content substantial for learning a given topic? Is the length of the module appropriate?
- [x] **Pedagogy:** Does the module seem easy to follow? Does it observe guidance on cognitive load? (working memory limits of 7 +/- 2 chunks of information)
- [x] **Content quality:** Is the writing of good quality, concise, engaging? Are the code components well crafted? Does the module seem complete?
- [x] **Instructional design:** Is the instructional design deliberate and apparent? For example, exploit worked-example effects; effective multi-media use; low extraneous cognitive load.

### JOSE paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state the need for this module and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

**Comment by whedon** [comment]
Hello human, I'm @whedon. I'm here to help you with some common editorial tasks. @oliviaguest, it looks like you're currently assigned as the reviewer for this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/jose-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/jose-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[--> Check article proof :page_facing_up: <--](https://github.com/openjournals/jose-papers/blob/jose.00016/jose.00016/10.21105.jose.00016.pdf)

**Comment by labarba** [comment]
👋 @oliviaguest, @ThomasA — Thank you for agreeing to review this submission to JOSE. Have a look at the [Reviewer Guidelines](http://jose.theoj.org/about#reviewer_guidelines), and feel free to ask any questions here.

Each of you has a reviewer checklist at the top of this issue thread. You should check off each item, as you complete your review. You're also encouraged to open new issues on the submission's repository, as needed, adding a reference to them here.

**Comment by oliviaguest** [comment]
Hi @arokem @labarba — I am not sure how to go about this, but if I understand correctly all items in my checklist are present/addressed except the final one. My issue is here: https://github.com/arokem/scipy-optimize/issues/4

**Comment by labarba** [comment]
You've gone about it precisely as we want 👍

**Comment by ThomasA** [comment]
@labarba, is this a software submission or a learning module submission ([your reviewer guidelines](https://jose.theoj.org/about#reviewer_guidelines))? I assume a learning module?

**Comment by ThomasA** [comment]
If it's OK by you I am just going to comment here bit by bit as I go through the review.

**Comment by ThomasA** [comment]
- Authorship: @arokem has made substantial contributions to the repository and is the sole author listed on the paper. A number of other contributors have made relatively minor contributions; I guess it is OK that these are not authors?

  @gvwilson is also a substantial contributor to the repository - in fact by far the major contributor. At his point it is not clear to me which parts of the repository he contributed and whether he ought to be a co-author on the paper as well. Can @arokem and @gvwilson please clarify?

**Comment by ThomasA** [comment]
- Example usage: as I see it, we can inspect this and several other aspects on two levels:
  - Being a teaching module, one level is examples given to students of the module when taught.
  - The other level is examples of using the teaching module for other potential teachers.

  I assume the second level (meta-level) is the most important in this review procedure.

  As such, I do not think this submission provides much guidance in terms of examples of how to actually use the module. Here I am thinking that it might not for example be immediately obvious to new adopters of the module that the content seems to be set up to generated through GitHub's jekyll and how to use that. In particular, the file [instructors.md](https://github.com/arokem/scipy-optimize/blob/gh-pages/instructors.md) in the repository is empty - except for some headlines. Perhaps the module follows some common structure from Software Carpentry that has some generic usage documentation for instructors that could be referenced?

**Comment by gvwilson** [comment]
The content is all @arokem - my contributions were solely to the Jekyll template.

**Comment by oliviaguest** [comment]
@ThomasA have you seen the materials at the repo's URL? http://arokem.github.io/scipy-optimize/

**Comment by ThomasA** [comment]
@oliviaguest yes, I guess that can be seen as "getting started" advice for both learners and instructors alike, but I was thinking if there should be more "meta-instruction" on which practicalities are involved if you want to clone and modify the material for your own purposes? It could be a simple matter of making potential instructors aware that the material relies on GitHub jekyll for generation and referring to GitHub's documentation for that. Or maybe Software Carpentry has some generic instructions for their material  that applies here?

**Comment by ThomasA** [comment]
Apart from my above comment on usage examples, that I would like to discuss, I consider my review complete and can recommend publication.

**Comment by oliviaguest** [comment]
Ah, I see what you mean. I agree that's a very useful idea.

**Comment by labarba** [comment]
I agree with @ThomasA. The Author Guidelines say:

> Computational learning modules should be complete and immediately usable for self-learning or adoption by other instructors.

In the paper (and possibly also the documentation), JOSE authors should explain how an instructor might adapt the module or how an independent learner might use it. 

Readers also want to know _why_ they might adopt/use the module. This has to do with the approach to teaching a topic, and the learning scenario.

(Bear in mind, we're looking at the first few submissions to JOSE, and we are solidifying our genre as we hold these very conversations.)

**Comment by arokem** [comment]
Indeed, I used the Carpentries (then Software Carpentry) lesson template to
set up the pages for this lesson.

I added a link to the Carpentries example lesson in the README in
https://github.com/arokem/scipy-optimize/commit/f6b2ae9dbded8e2610900292a439bf6e369c9a8a

On Thu, Jun 28, 2018 at 5:53 AM Lorena A. Barba <notifications@github.com>
wrote:

> I agree with @ThomasA <https://github.com/ThomasA>. The Author Guidelines
> say:
>
> Computational learning modules should be complete and immediately usable
> for self-learning or adoption by other instructors.
>
> In the paper (and possibly also the documentation), JOSE authors should
> explain how an instructor might adapt the module or how an independent
> learner might use it.
>
> Does that address this issue?

> Readers also want to know *why* they might adopt/use the module. This has
> to do with the approach to teaching a topic, and the learning scenario.
>
> I don't know what this means. Could you give an example (e.g., from
another paper) demonstrating this?

**Comment by labarba** [comment]
I see that only two check-boxes remain unticked in the review checklists.

Are we waiting on a couple of improvements from the author, at this point?

**Comment by ThomasA** [comment]
I think linking to the Software Carpentry lesson introduction has helped http://carpentries.github.io/lesson-example/. The submission is OK by me now.

**Comment by labarba** [comment]
@arokem Tiny fix: `SciPy` has a capital P (paper).

**Comment by labarba** [comment]
@arokem More fixes:

* Summary: In the reference to Jones at al. 2001, we see the year twice in the citation.
* Statement of need: 
   * "such as 'Software Carpentry'," --> place the comma inside the quote.
   * "The target audience ... are researchers" --> *is*
* Learning objectives: "In addition to these" --> *the*
* Description of the module: 
   * "The core of the course are" --> *is*
   * "to fit these functional form" --> *forms* (or *the* instead of *these*)
   *  "sum of square error objective function" --> *sum-of-square-error* (hyphens for compound adjective)

**Comment by oliviaguest** [comment]
Hi @labarba I'm ticking it on the proviso that this (https://github.com/arokem/scipy-optimize/issues/4) is all sorted, which I assume it is?

**Comment by arokem** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[--> Check article proof :page_facing_up: <--](https://github.com/openjournals/jose-papers/blob/jose.00016/jose.00016/10.21105.jose.00016.pdf)

**Comment by arokem** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[--> Check article proof :page_facing_up: <--](https://github.com/openjournals/jose-papers/blob/jose.00016/jose.00016/10.21105.jose.00016.pdf)

**Comment by labarba** [comment]
Great, @arokem — I see all the changes, and your paper is ready to accept!

Please make an archive now on your chosen repository, and tell us the DOI.

**Comment by arokem** [comment]
@labarba : do you understand why the year 2001 appears twice in the rendered pdf? It only appears once in [the bib file](https://github.com/arokem/scipy-optimize/blob/gh-pages/paper.bib#L11-L16)

And if I remove that, I get no year at all... (see the recent compilation attempt). Maybe whedon doesn't know how to properly handle the "misc" bib category?

**Comment by labarba** [comment]
@oliviaguest, @ThomasA — Thank you both for volunteering to review this submission, and being part of the JOSE adventure!

**Comment by arokem** [comment]
Thanks all! Here is the Zenodo DOI for the archive: 10.5281/zenodo.1304473

**Comment by labarba** [comment]
@whedon set 10.5281/zenodo.1304473 as archive

**Comment by whedon** [comment]
OK. <a href="http://dx.doi.org/10.5281/zenodo.1304473" target="_blank">10.5281/zenodo.1304473</a> is the archive.

**Comment by labarba** [comment]
@arfon, I wonder if you could help here. We're having a little problem with one of the references. See the comment by @arokem, [above](https://github.com/openjournals/jose-reviews/issues/16#issuecomment-402325346).

**Comment by labarba** [comment]
@arokem Why do you put two dashes after the year?

**Comment by arfon** [comment]
We had to this the other day with another bibtex file for an `online` entry
type. I’m afraid I don’t know what the fix is here (we didn’t figure it out
on the other submission)

**Comment by arokem** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[--> Check article proof :page_facing_up: <--](https://github.com/openjournals/jose-papers/blob/jose.00016/jose.00016/10.21105.jose.00016.pdf)

**Comment by arokem** [comment]
Yep, removing the dashes does remove the year duplication. Should I create a new archive for this change?

**Comment by labarba** [comment]
Can you just up the version of the archive on Zenodo? (keeps the DOI)

**Comment by labarba** [comment]
@arokem Note that Zenodo grabs the authors automatically from the GitHub repo. You need to manually change the author list there, since you get people listed that are not authors of this module.

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippet:

```
[![DOI](https://jose.theoj.org/papers/10.21105/jose.00016/status.svg)](https://doi.org/10.21105/jose.00016)
```
This is how it will look in your documentation:

[![DOI](https://jose.theoj.org/papers/10.21105/jose.00016/status.svg)](https://doi.org/10.21105/jose.00016)

**We need your help!**

Journal of Open Source Education is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. 
- Making a small donation to support our running costs here: https://www.flipcause.com/secure/cause_pdetails/Mjk3ODA=

**Comment by labarba** [comment]
@arokem Your paper is now published, yippee!!!  
Please sign up as reviewer 😁 

@oliviaguest, @ThomasA — Thank you again for reviewer. Do sign up (if you haven't) to our official reviewer list, for future requests, and help us advertise JOSE!

**Comment by arokem** [comment]
Thank you! Yippee!
On Wed, Jul 4, 2018 at 12:38 PM Lorena A. Barba <notifications@github.com>
wrote:

> @arokem <https://github.com/arokem> Your paper is now published, yippee!!!
> Please sign up as reviewer 😁
>
> <https://github.com/oliviaguest>
>
I believe I already did!

**Comment by labarba** [comment]
:loudspeaker:  New Reviewer Sign-up Form!
https://forms.gle/Rizd3TcHnQKhrbYY7
